Userland/Libraries/LibWeb/WebIDL/AbstractOperations.cpp

```diff
@@ -97,12 +100,15 @@ inline JS::Completion clean_up_on_return(HTML::EnvironmentSettingsObject& stored
         return completion;
 
     // 4. If completion is an abrupt completion and the operation has a return type that is not a promise type, return completion.
-    // FIXME: This does not handle promises and thus always returns completion at this point.
-    return completion;
+    if (completion.is_abrupt() && operation_returns_promise == OperationReturnsPromise::No)
+        return completion;
 
-    // FIXME: 5. Let rejectedPromise be ! Call(%Promise.reject%, %Promise%, «completion.[[Value]]»).
+    // 5. Let rejectedPromise be ! Call(%Promise.reject%, %Promise%, «completion.[[Value]]»).
+    auto rejected_promise = create_rejected_promise(realm, *completion.release_value());
```

Comment:
Curious that https://webidl.spec.whatwg.org/#a-promise-rejected-with even exists given it does the exact same thing as `%Promise.reject%`...

Reply:
Yeah, I guess from a spec author perspective, "create a promise rejected with \<value\>" is easier to type than the `Call` syntax (but then why is the `Call` syntax used here :sweat_smile:)